staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -431,14 +432,36 @@ func (a *Admission) EvaluatePodsInNamespace(ctx context.Context, namespace strin
 	var warnings []string
 	if len(pods) > namespaceMaxPodsToCheck {
 		warnings = append(warnings, fmt.Sprintf("Large namespace: only checking the first %d of %d pods", namespaceMaxPodsToCheck, len(pods)))
-		pods = pods[0:namespaceMaxPodsToCheck]
 	}
 
+	var skippedPods []*corev1.Pod
+	totalEvaluatedPods := 0
+	evaluatedControllers := make(map[types.UID]bool)
+	skipPod := false
 	for i, pod := range pods {
+		skipPod = false
+		if totalEvaluatedPods == namespaceMaxPodsToCheck {
+			break
+		}
+
 		// short-circuit on exempt runtimeclass
 		if a.exemptRuntimeClass(pod.Spec.RuntimeClassName) {
 			continue
 		}
+
+		// short-circuit if pod from the same controller is evaluated
+		for _, podOwnerRef := range pod.GetOwnerReferences() {
+			if _, ok := evaluatedControllers[podOwnerRef.UID]; !ok {
```

Comment:
nit: since this is a map to bool
```suggestion
			if evaluatedControllers[podOwnerRef.UID] {
```

Reply:
Makes sense, Habit got the better of me